### ISSUE TITLE
Using inject() issues

### ISSUE THREAD

**Issue by dlgoodchild** [jeffijoe/awilix-koa#65]
Hi, before I go further down the rabbit hole, is `inject()` definitely working as intended?

So I'm using it like so: 
```
@before([authViaPassportJson, inject(authUpdateLastLoginFactory)])
```

This then causes the following error:
```
[1] ERROR:
[1] TypeError: resolved.apply is not a function
[1]     at middlewareFactoryHandler (.../api/node_modules/awilix-koa/src/invokers.ts:119:2)
[1]     at dispatch (.../api/node_modules/koa-compose/index.js:42:32)
[1]     at .../api/node_modules/koa-passport/lib/framework/koa.js:160:16
[1]     at processTicksAndRejections (node:internal/process/task_queues:95:5)
[1]     at async authViaPassportJson (.../api/src/module/app/auth/auth.middleware.ts:48:3)
````
What's weird about this is that `authViaPassportJson` isn't a factory and isn't using awilix. I was pulling my hair out at this as it just didn't make any sense whatsoever. I even reduced that middleware down to effectively `return true`.

I then changed that `@before` to make it simpler and to focus, to just:
```
@before(authViaPassportJson)
```
This works, no problem. So I turned my attention to `inject(authUpdateLastLoginFactory)`. This is where the true issue lies.
If I remove the dependency requirement, simplify the internals and not use `inject` then it works. Again, narrowing down the issue.

On using just:
```
@before(inject(authUpdateLastLoginFactory))
```
I then get:
```
[1] ERROR:
[1] TypeError: resolved.apply is not a function
[1]     at middlewareFactoryHandler (.../api/node_modules/awilix-koa/src/invokers.ts:119:2)
[1]     at dispatch (.../api/node_modules/koa-compose/index.js:42:32)
[1]     at .../api/node_modules/koa-router/lib/router.js:425:16
[1]     at dispatch (.../api/node_modules/koa-compose/index.js:42:32)
[1]     at .../api/node_modules/koa-unless/index.js:26:14
[1]     at dispatch (.../api/node_modules/koa-compose/index.js:42:32)
[1]     at .../api/node_modules/koa-router/lib/router.js:425:16
[1]     at dispatch (.../api/node_modules/koa-compose/index.js:42:32)
[1]     at .../api/node_modules/koa-unless/index.js:26:14
[1]     at dispatch (.../api/node_modules/koa-compose/index.js:42:32)
```
Which is a somewhat unexpected callstack.

To sum up. I'm using `makeControllers`, loading a load of singleton services, daos and such-like, all without issue. The moment I introduce `inject` anywhere, it falls over. I've tested a myriad of middlewares using inject (basically turned them all into factories) and I've not succeeded in getting any to work.

Like I say, before going down the rabbit hole further, I'd like to get a sense of if `inject` has been tested recently with koa...or maybe I'm just not using it correctly?

Cheers.

**Comment by dlgoodchild** [comment]
Worked it out. When refactoring my code, I had left in the `async` on the factories.
I'm an idiot! Spent ages debugging this!
